config/utilities/uenv.py

```diff
@@ -13,6 +13,29 @@
 _UENV_MOUNT_DELIMITER = '@'
 _RFM_META = pathlib.Path('extra') / 'reframe.yaml'
 
+# returns one of
+#   'a100', 'gh200', 'mi200', 'zen2', 'zen3'
+#   None
+def uarch(partition):
```

Comment:
I would put this in a separate file since it might be useful outside uenv

Reply:
The `uarch` labels are specific to uenv (they correspond to the node types on Alps, and images are labelled with them).

https://confluence.cscs.ch/display/VCUE/UENV

So I would prefer to keep them in the `uenv` "namespace".